### PR TITLE
Improve support for XDG on MacOS

### PR DESCRIPTION
**Pull request by SolitudeSF**
Im not familiar with rust, so its just an attempt to improve the situation.

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -240,6 +240,22 @@ pub struct App {
...
+fn get_config_dir() -> Option<PathBuf> {
+    dirs::config_dir()
+}
+
+#[cfg(target_os = "macos")]
+fn get_config_dir() -> Option<PathBuf> {
+    match env::var_os("XDG_CONFIG_HOME") {
+        Some(val) => Some(PathBuf::from(val)),
+        None => match dirs::home_dir() {
+            Some(val) => Some(val.join("/Library/Application Support")),
```
Comment (sayanarijit) [inline src/app.rs:253]:
```suggestion
            Some(val) => Some(val.join(".config")),
```
So we don't break existing config files.

**Review by sayanarijit — CHANGES_REQUESTED** [review]
Let's keep it as a potential solution, but let's also try alternatives.

**Review by sayanarijit — APPROVED** [review]

**Comment by sayanarijit** [comment]
@noahmayr what do you think of it as a potential solution for https://github.com/sayanarijit/xplr/issues/627?

**Comment by noahmayr** [comment]
It does seem to work (on MacOS), tested both with and without XDG_CONFIG_HOME. However I feel like using the xdg-rust crate you suggested in #627 might be better to avoid platform specific logic in xplr.

Does xdg-rust cover all usecases where dirs is used for right now?

**Comment by sayanarijit** [comment]
Agree. Less platform specific code is preferable.

**Comment by sayanarijit** [comment]
Merged https://github.com/sayanarijit/xplr/pull/631 since as per the doc, it shouldn't break existing config. Will re-open this pr if it does.

**Pull request closed without merging**